tests/unit-tests/src/VuFindTest/OaiPmh/CommunicatorTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * OAI-PMH harvester factory integration test.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2010.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development
+ */
+namespace VuFindTest\Harvest\OaiPmh;
+
+use Laminas\Http\Client;
+use VuFindHarvest\OaiPmh\Communicator;
+
+/**
+ * OAI-PMH communicator test.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development
+ */
+class CommunicatorTest extends \PHPUnit\Framework\TestCase
+{
+
+    /**
+     * Get Communicator
+     *
+     * @param string                     $uri       Base URI for OAI-PMH server
+     * @param Client                     $client    HTTP client
+     *
+     * @return type
+     */
+    protected function getCommunicator($uri, $client)
+    {
+        return new Communicator($uri, $client);
+    }
+
+    /**
+     * Test a simple communicator request.
+     *
+     * @return void
+     */
+    public function testSimpleRequest()
+    {
+        $client = $this->getMockClient();
+        $response = $client->send();
+        $response->expects($this->any())
+            ->method('isSuccess')
+            ->will($this->returnValue(true));
+        $response->expects($this->any())
+            ->method('getBody')
+            ->will($this->returnValue($this->getIdentifyResponse()));
+        $uri = 'http://localhost';
+        $comm = $this->getCommunicator($uri, $client);
+        $this->assertEquals(
+            $comm->request('Identify'),
```

Comment:
You should reverse these parameters -- first is "expected" and second is "actual".

Reply:
Maybe it would be slightly more clear if you set `$expectedResponse = $this->getIdentifyResponse();` earlier in the test and use that instead of calling the method twice (but I'll leave that to your discretion).

Reply:
Whoops, a silly mistake there. Also, good point about establishing one variable.